GGRC-191 Preselect the First Assessment Template when generating Assessments

plamut opened the pull request:
This PR changes the generate Assessments modal, so that by default, the first available AssessmentTemplate is selected in the corresponding dropdown.

**Details:**
- Open an Audit, go to the Assessment Templates tab.
- If needed, define a couple of Assessment Templates for different object types (some of the ATs should have the same object type set).
- Go to  the Assessments tab and click the magic wand icon (just above the tree view, on the right).

**Expected result:**
In the modal, the first available Assessment Template should be preselected in the corresponding dropdown (if such template exists, of course)

**Actual result:**
No Assessment Template is selected in the dropdown by default.

P.S.: The branch name is incorrect, the issue number is indeed GGRC-191.

j0gurt commented on src/ggrc/assets/javascripts/components/assessment_templates/assessment_templates.js:
```diff
@@ -44,15 +46,57 @@
+          return;
+        }
+
+        nonDummyItem = templates[1];  // a single item or an object group
```
Not so good to get exactly second element from the array, will be better put inside the loop and filter by value

plamut replied:
Can you elaborate please? The first option is always a dummy option, thus we can be sure that the first one that follows is the one that we want.

plamut commented:
Temporarily added `on hold` to not merge anything into the release branch today.

j0gurt commented on src/ggrc/assets/mustache/modals/mapper/base.mustache:
```diff
@@ -73,6 +73,7 @@
     type="mapper.type"
     binding="related_assessment_templates"
     assessment-template="mapper.assessmentTemplate"
+    mapper="mapper"
```
You inserted the whole object into component so no needed additional attributes here

plamut replied:
True that. I didn't change that, though, because that felt out of scope of this PR (i.e. refactoring the components compared to only fixing an issue).

If you nevertheless think this should be included, I will make the change, although personally I would do it in a separate PR. It's up to you, the peers, to make a verdict. :)

j0gurt replied:
@plamut we need to not forget do it in future

plamut replied:
Definitely. I'll make sure that a ticket gets created.

plamut commented:
@j0gurt I have updated the PR.The refactoring ticket will be opened by the QA team, and I added a console warning if a potential API error is detected.

I'm hence removing  the `needs work` label, but if you think there is more to to, please add it back, thanks.

j0gurt commented:
@plamut cool thanks
